src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -150,15 +151,23 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                     catch (Exception ex)
                     {
                         _fetchMetadataFailure = ex;
-                        _syncAfter = DateTimeUtil.Add(now.UtcDateTime, AutomaticRefreshInterval < RefreshInterval ? AutomaticRefreshInterval : RefreshInterval);
-                        if (_currentConfiguration == null) // Throw an exception if there's no configuration to return.
+
+                        if (_currentConfiguration == null)
+                        {
+                            _syncAfter = DateTimeUtil.Add(now.UtcDateTime, new[] { AutomaticRefreshInterval, RefreshInterval, BootstrapRefreshInterval }.Min());
```

Comment:
If RefreshInterval < BootstrapRefreshInterval, then the Min() will be less than BootstrapRefreshInterval.

For example, if RefreshInterval = 5 seconds, then _syncAfter will be now + 5 sec.

Should the logic be Math.Max(Math.Min(AutomaticRefreshInterval, RefreshInterval), BootstrapRefreshInterval)?

Reply:
If this is to allow retrieving metadata (again) in less than 5 min at the startup when the first attempt fails, can't users just set the RefreshInterval to 30 seconds instead of adding the new BootstrapRefreshInterval?

Reply:
Should we do this instead?
_syncAfter = now.UtcDateTime + BootstrapRefreshInterval;

Reply:
resolved